Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,176 @@
+
+            var name = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                Console.Write("Name cannot be empty.");
+                return "-";
+            }
+            else
+            {
+                return name;
+            }
+        }
+        
+        public static int PromptAgeInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+            string age = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(age))
+            {
+                return 0;
+            }
+            else if (IsValidPositiveNumber(age))
+            {
+                return int.Parse(age);             
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(age);
+                return -1;
+            }
+        }
+
+        internal static bool IsValidPositiveNumber(string value) => IsNumber(value) && IsPositive(value);
+
+        public static float PromptFloatInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+
+            var value = Console.ReadLine();
+            if (IsNumber(value) && IsPositive(value))
+            {
+                return float.Parse(value);
+            }
+            else if(string.IsNullOrEmpty(value))
+            {
+                return 0;
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(value);
+                return -1;
+            }
+        }
+       
+        public static bool ValidateHeightAndWeight(float weight, float height)
+        {
+            bool isValidWeight = weight > 0, isValidHeight = height > 0;
+
+            return ((isValidHeight && isValidWeight) && (weight > height));
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+
+            bool isHeightAndWeightValid = ValidateHeightAndWeight(weight, height);
+
+            if (isHeightAndWeightValid)
+            {
+                return weight / MathF.Pow(height, 2);
+            }
+            else
+            {
+                if (weight <= 0 && height <= 0)
+                {
+                    DisplayErrorMessageForNumbers($"Weight cannot be equal or less than zero, but was {weight}.{Environment.NewLine}Height cannot be less than zero, but was {height}.");
+                }
+                else if (height <= 0)
+                {
+                    DisplayErrorMessageForNumbers($"Height cannot be equal or less than zero, but was {height}.");
+                }else if(weight >= height)
+                {
+                    DisplayErrorMessageForNumbers($"Weight cannot be more or equal to height.{Environment.NewLine} Height= {height}, Weight= {weight}.");
+                }else
+                {
+                    DisplayErrorMessageForNumbers($"Weight cannot be equal or less than zero, but was {weight}.");
+                }
+                return -1;
```

Comment:
Here you have quite some duplicated checking. Not much impact though and possible easier to read/understand.